pkg/model/iam.go

```diff
@@ -103,26 +104,63 @@ func (b *IAMModelBuilder) Build(c *fi.ModelBuilderContext) error {
 		}
 	}
 
+	irsa := b.Cluster.Spec.IAMRolesForServiceAccounts
+	if irsa != nil {
+		for _, sa := range irsa.ServiceAccounts {
+			var p *iam.Policy
+			if sa.InlinePolicy != "" {
+				bp, err := b.buildPolicy(sa.InlinePolicy)
+				p = bp
+				if err != nil {
+					return fmt.Errorf("error inline policy: %w", err)
+				}
+			}
+			serviceAccount := &iam.GenericServiceAccount{
+				NamespacedName: types.NamespacedName{
+					Name:      sa.Name,
+					Namespace: sa.Namespace,
+				},
+				Policy: p,
+			}
+			iamRole, err := b.BuildServiceAccountRoleTasks(serviceAccount, c)
+			if err != nil {
+				return fmt.Errorf("error building service account role tasks: %w", err)
+			}
+			if sa.IAMPolicyARN != "" {
+				name := "external-" + fi.StringValue(iamRole.Name)
+				externalPolicies := []string{sa.IAMPolicyARN}
+				c.AddTask(&awstasks.IAMRolePolicy{
```

Comment:
Would we ever want to support attaching multiple policy ARNs to a single role? if so we should make that api field a list and include a reference to the policy in the `name` here.

Reply:
That's a good point. I don't see a reason why we shouldn't allow that.